### PR TITLE
Mm 33069/move group members to schema migrations

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
implements: https://mattermost.atlassian.net/browse/MM-33069

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33069

#### Release Note

```release-note
* Moves Groups gorp driven migrations to schema-based migrations
```

### CODE REVIEW

File: db/migrations/mysql/000003_create_user_groups.up.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'UserGroups'
+        AND table_schema = DATABASE()
+        AND column_name = 'AllowReference'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE UserGroups ADD AllowReference tinyint(1);'
```
Comment (mgdelacroix) [inline db/migrations/mysql/000003_create_user_groups.up.sql:27]: This is created with a default value of 0, should we include it here?

```go
sqlStore.CreateColumnIfNotExists("UserGroups", "AllowReference", "boolean", "boolean", "0")
```

File: db/migrations/mysql/000003_create_user_groups.up.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
```
Comment (mgdelacroix) [inline db/migrations/mysql/000003_create_user_groups.up.sql:34]: Did those two failed in the CI checks as well or are we defaulting to create indexes both in the table and after for MySQL?

File: db/migrations/mysql/000004_create_group_members.up.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+CREATE TABLE IF NOT EXISTS GroupMembers (
+    GroupId varchar(26) NOT NULL,
+    UserId varchar(26) NOT NULL,
+    CreateAt bigint(20) DEFAULT NULL,
+    DeleteAt bigint(20) DEFAULT NULL,
+    PRIMARY KEY (GroupId, UserId),
+    KEY idx_groupmembers_create_at (CreateAt)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
+
+SET @preparedStatement = (SELECT IF(
```
Comment (mgdelacroix) [inline db/migrations/mysql/000004_create_group_members.up.sql:10]: Same index question with this one

File: db/migrations/mysql/000005_create_group_teams.up.sql [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'GroupTeams'
+        AND table_schema = DATABASE()
+        AND column_name = 'SchemeAdmin'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE GroupTeams ADD COLUMN SchemeAdmin tinyint(1);'
```
Comment (mgdelacroix) [inline db/migrations/mysql/000005_create_group_teams.up.sql:22]: Same default question for this one:

```go
sqlStore.CreateColumnIfNotExists("GroupTeams", "SchemeAdmin", "boolean", "boolean", "0")
```

File: db/migrations/postgres/000004_create_group_members.up.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE IF NOT EXISTS groupmembers (
+groupid VARCHAR(26),
+userid VARCHAR(64),
+createat bigint,
+deleteat bigint,
+PRIMARY KEY(groupid, userid)
```
Comment (mgdelacroix) [inline db/migrations/postgres/000004_create_group_members.up.sql:6]:
```suggestion
    groupid VARCHAR(26),
    userid VARCHAR(64),
    createat bigint,
    deleteat bigint,
    PRIMARY KEY(groupid, userid)
```

File: db/migrations/mysql/000003_create_user_groups.up.sql (hunk 1 above)
Comment (nronas) [reply db/migrations/mysql/000003_create_user_groups.up.sql:27]: interesting, the schema generated is the same, though. Will put the default in for readabillity.

**Review by mgdelacroix — APPROVED** [review]

**Pull request merged**